packages/console/src/monaco/MonacoProviders.tsx

```diff
@@ -0,0 +1,346 @@
+/**
+ * Completion provider for a code session
+ */
+import { PureComponent } from 'react';
+import * as monaco from 'monaco-editor';
+import Log from '@deephaven/log';
+import { IdeSession } from '@deephaven/jsapi-shim';
+
+const log = Log.module('MonacoCompletionProvider');
+
+interface MonacoProviderProps {
+  model: monaco.editor.ITextModel;
+  session: IdeSession;
+  language: string;
+}
+
+/**
+ * Registers a completion provider with monaco for the language and session provided.
+ */
+class MonacoProviders extends PureComponent<
+  MonacoProviderProps,
+  Record<string, never>
+> {
+  /**
+   * Converts LSP CompletionItemKind to Monaco CompletionItemKind
```

Comment:
Can you add a comment linking to the spec? https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItemKind

I'm guessing there's no LSP constants we can use from somewhere?